skimage/io/collection.py

```diff
@@ -256,10 +255,15 @@ def __getitem__(self, n):
                 if self._frame_index:
                     fname, img_num = self._frame_index[n]
                     if img_num is not None:
-                        self.data[idx] = self.load_func(fname, img_num=img_num,
-                                                        **kwargs)
-                    else:
+                        kwargs['img_num'] = img_num
+                    try:
                         self.data[idx] = self.load_func(fname, **kwargs)
+                    except TypeError as e:
```

Comment:
I would add a comment on the intention behind this; that's the kind of things hard to read afterwards